<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>580</width>
    <height>599</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>580</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#frame {
	background-color: white;
	border-radius: 5px;
}

QTabBar::tab {
	width: 80px;
}

QTabBar::tab:selected {
	font-weight: bold;
	
}

QTabBar::tab:!selected {
	color: gray;
	border: none;
}

QTabBar::tab:top:!selected {
    margin-top: 3px;
}

QTabBar::tab:bottom:!selected {
    margin-bottom: 3px;
}

QTabBar::tab:!selected:hover {
    background: lightgray;
}

#about_description {
	margin-bottom: 20px;
}

#log_console {
	background-color: #1b232f;
	border-radius: 5%;
	color: #ebebeb;
	padding: 10px;
}

#client_status, #server_status, #client_data, #server_data {
	border: 1px solid silver;
	background-color: #fff;
}

#client_label, #server_label, #client_label_2, #server_label_2 {
	border: 1px solid silver;
	background-color: #e8e9eb;
	padding: 10px;
}

#connection_note, #data_note {
	border: 1px solid silver;
	border-left: 5px solid #4070f4;
	padding: 10px;
	background-color: #f5f5f5;
}

#usability_title {
	font-size: 11pt;
	font-weight: bold;
	margin-bottom: 20px;
}

#measure_bar {
	min-height: 6px;
    max-height: 6px;
    border-radius: 3px;
	background-color: #d4d4d5;
}

#measure_bar::chunk {
	min-height: 6px;
    max-height: 6px;
    border-radius: 3px;
	background-color: #0b69e5;
}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="cursor">
         <cursorShape>ArrowCursor</cursorShape>
        </property>
        <property name="tabShape">
         <enum>QTabWidget::Rounded</enum>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="about_panel">
         <attribute name="title">
          <string>About</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item alignment="Qt::AlignTop">
           <widget class="QLabel" name="about_description">
            <property name="text">
             <string/>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="usability_title">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Usability</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="measure_bar">
            <property name="value">
             <number>0</number>
            </property>
            <property name="textVisible">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="usability_form" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7"/>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="log_panel">
         <attribute name="title">
          <string>Log</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPlainTextEdit" name="log_console">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="status_panel">
         <attribute name="title">
          <string>Status</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QWidget" name="client_status" native="true">
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="client_label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="styleSheet">
                <string notr="true">font-size: 11pt;</string>
               </property>
               <property name="text">
                <string>Client</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QGridLayout" name="gridLayout">
               <property name="leftMargin">
                <number>20</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="bottomMargin">
                <number>10</number>
               </property>
               <property name="horizontalSpacing">
                <number>10</number>
               </property>
               <property name="verticalSpacing">
                <number>20</number>
               </property>
               <item row="1" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="maximumSize">
                  <size>
                   <width>130</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Request</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="maximumSize">
                  <size>
                   <width>130</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Connection</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Established</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLabel" name="request_status">
                 <property name="text">
                  <string>NULL</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="maximumSize">
                  <size>
                   <width>130</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Timestamp</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLabel" name="timestamp_status">
                 <property name="text">
                  <string>NULL</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="server_status" native="true">
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="server_label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="styleSheet">
                <string notr="true">font-size: 11pt;</string>
               </property>
               <property name="text">
                <string>Server</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0">
               <property name="leftMargin">
                <number>20</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="bottomMargin">
                <number>10</number>
               </property>
               <property name="horizontalSpacing">
                <number>10</number>
               </property>
               <property name="verticalSpacing">
                <number>20</number>
               </property>
               <item row="2" column="1">
                <widget class="QLabel" name="response_status">
                 <property name="text">
                  <string>NULL</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label_9">
                 <property name="maximumSize">
                  <size>
                   <width>130</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Status</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_11">
                 <property name="maximumSize">
                  <size>
                   <width>130</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Response</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="server_status_message">
                 <property name="text">
                  <string>OK</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_13">
                 <property name="maximumSize">
                  <size>
                   <width>130</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Status Code</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLabel" name="server_status_code">
                 <property name="text">
                  <string>200</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QScrollArea" name="scrollArea_3">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>509</width>
               <height>250</height>
              </rect>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_15">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="connection_note">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>&lt;p&gt;&lt;b&gt;Eavesdropping Attacks&lt;/b&gt;: In an eavesdropping attack, an attacker passively listens to the communication between two parties to gather sensitive information. This is why encryption is important in secure communication.&lt;/p&gt;

&lt;p&gt;&lt;b&gt;Replay Attacks&lt;/b&gt;: In a replay attack, an attacker intercepts the data and retransmits it later. This can be prevented by using time-stamped tokens or one-time passwords.&lt;/p&gt;

&lt;p&gt;&lt;b&gt;Session Hijacking&lt;/b&gt;: In a session hijacking attack, an attacker takes over a session between a client and server after the user has successfully authenticated. This allows the attacker to do anything the user can do on their account.&lt;/p&gt;

&lt;p&gt;&lt;b&gt;IP Spoofing&lt;/b&gt;: In an IP spoofing attack, an attacker sends messages to a computer with an IP address indicating that the message is coming from a trusted host. This can be used to gain unauthorised access to a machine.&lt;/p&gt;</string>
                </property>
                <property name="wordWrap">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="data_panel">
         <attribute name="title">
          <string>Data</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QWidget" name="client_data" native="true">
            <layout class="QVBoxLayout" name="layout_2">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="client_label_2">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="styleSheet">
                <string notr="true">font-size: 11pt;</string>
               </property>
               <property name="text">
                <string>Client</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QGridLayout" name="client_data_layout">
               <property name="leftMargin">
                <number>20</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="bottomMargin">
                <number>10</number>
               </property>
               <property name="spacing">
                <number>20</number>
               </property>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="server_data" native="true">
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="server_label_2">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="styleSheet">
                <string notr="true">font-size: 11pt;</string>
               </property>
               <property name="text">
                <string>Server</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QGridLayout" name="server_data_layout" rowstretch="0">
               <property name="leftMargin">
                <number>20</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="bottomMargin">
                <number>10</number>
               </property>
               <property name="spacing">
                <number>20</number>
               </property>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="data_note_container" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_12">
             <property name="spacing">
              <number>11</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>11</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QScrollArea" name="scrollArea_4">
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="widgetResizable">
                <bool>true</bool>
               </property>
               <widget class="QWidget" name="scrollAreaWidgetContents_2">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>100</width>
                  <height>38</height>
                 </rect>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_16">
                 <property name="spacing">
                  <number>0</number>
                 </property>
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="data_note">
                   <property name="text">
                    <string/>
                   </property>
                   <property name="wordWrap">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
